superset/security/guest_token.py

```diff
@@ -0,0 +1,87 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from enum import Enum
+from typing import List, Optional, TypedDict, Union
```

Comment:
@suddjian `TypedDict` was introduced in Python 3.8 ( https://www.python.org/dev/peps/pep-0589/ , https://docs.python.org/3/whatsnew/3.8.html#typing ). The minimal supported version of Python by Superset is 3.7 ( https://github.com/apache/superset/blob/master/setup.py#L173-L175 ). We have to express that in a different way or drop support for Python 3.7.

Reply:
I would like add that it make pre-commit hook `mypy` fail on Python 3.7:
```
superset/security/guest_token.py:18: error: Module "typing" has no attribute "TypedDict"; maybe "_TypedDict"?
superset/security/guest_token.py:24: error: Unexpected keyword argument "total" for "__init_subclass__" of "object"
Found 2 errors in 1 file (checked 2 source files)
````

Reply:
Oh no, I'll push a fix. Sorry for the inconvenience.